### ISSUE TITLE
Navigation `<nav>` element labels should not contain the term 'navigation'

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#68682]
### Description

Screen readers already announce `<nav>` elements by using the term 'navigation' because that's the implicit role of a `<nav>` element.

Using the term 'navigation' in the `arira-label` of a `<nav>` element produces a double announcement of the term 'navigation':
1. As part of the accessible name.
2. As the element's role.

As such, all the `aria-label` attributes used to label any `<nav>` element should never contain the term 'navigation'.

This applies to both the Navigation block and any other `<nav>` element used in the editor.

The navigation block automatically generates a name for the navigation menu that is then used as the aria-label:

https://github.com/WordPress/gutenberg/blob/5a5455cb6f7633ececc00559b7a407b42f4874d6/packages/block-library/src/navigation/edit/use-generate-default-navigation-title.js#L51-L58

This should be changed to 'Menu' to avoid a redundant, noisy announcement. Same applies to other `<nav>` elements used in the editor.

The editor UI already refers to the concept of 'Menu' in many places (see screenshot below) so that appears to be a term users are already familiar with. The same screenshot illustrates how VoiceOver announces the menus:
- Highlighted in green: the announcement of the role.
- Note: when a `<nav>` element is empty, VoiceOver announces `empty navigation`.
- Highlighted in red: the default aria-label used by the editor.

![Image](https://github.com/user-attachments/assets/f88d288c-e0a1-4c8f-b3fc-78e754d54bdd)

### Step-by-step reproduction instructions

- Create a new navigation menu.
- Observe the default menu name and aria-label contain the term 'navigation'.
- Go to Styles > Revisions.
- Assuming there are many revisions, the panel will show pagination links.
- Observe the aria-label of the `<nav>` element is `Global styles pagination navigation'.

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure
